src/python/pants/engine/build_files.py

```diff
@@ -204,10 +204,10 @@ def _hydrate(item_type, spec_path, **kwargs):
 
 
 @rule
-async def provenanced_addresses_from_address_families(
+async def addresses_with_origins_from_address_families(
```

Comment:
```suggestion
async def addresses_with_origins_via_address_families(
```
This isn't super important, but it's not clear where the address families are coming from without looking at the body of the rule. I don't think the suggested change really helps with that, though, so probably just ignore this.